cdap-tms/src/main/java/co/cask/cdap/messaging/TopicMetadata.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.messaging;
+
+import co.cask.cdap.proto.id.TopicId;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents metadata about a messaging topic.
+ */
+public class TopicMetadata {
+
+  private static final String TTL_KEY = "ttl";
+
+  private final TopicId topicId;
+  private final Map<String, String> properties;
+
+  /**
+   * Creates a new instance for the given topic with the associated properties.
+   */
+  public TopicMetadata(TopicId topicId, Map<String, String> properties) {
+    validateProperties(topicId, properties);
+
+    this.topicId = topicId;
+    this.properties = ImmutableMap.copyOf(properties);
+  }
+
+  /**
+   * Creates a new instance for the given topic with the associated properties.
+   *
+   * @param topicId topic id
+   * @param properties a list of key/value pairs that will get converted into a {@link Map}.
+   */
+  @VisibleForTesting
+  public TopicMetadata(TopicId topicId, Object...properties) {
+    this(topicId, toMap(properties));
+  }
+
+  /**
+   * Returns the topic id that this metadata is associated with.
+   */
+  public TopicId getTopicId() {
+    return topicId;
+  }
+
+  /**
+   * Returns the raw properties for the topic.
+   */
+  public Map<String, String> getProperties() {
+    return properties;
+  }
+
+  /**
+   * Returns the time-to-live in seconds property of the topic.
+   */
+  public long getTTL() {
+    validateTTL(topicId, properties);
```

Comment:
Why do we have to validateTTL again (look like it is being done in the constructor)?